src/game/CreatureEventAI.cpp

```diff
@@ -1370,11 +1370,18 @@ void CreatureEventAI::UpdateAI(const uint32 diff)
         // Update creature dynamic movement position before doing anything else
         if (m_DynamicMovement)
         {
-            if (!m_creature->hasUnitState(UNIT_STAT_CAN_NOT_REACT) && !m_creature->IsNonMeleeSpellCasted(false))
+            if (!m_creature->hasUnitState(UNIT_STAT_CAN_NOT_REACT))
             {
-                if (m_LastSpellMaxRange && m_creature->IsInRange(m_creature->getVictim(), 0, (m_LastSpellMaxRange / 1.5f)))
-                    SetCombatMovement(false, true);
-                else
+                bool uiInLineOfSight = m_creature->IsWithinLOSInMap(m_creature->getVictim());
+                
+                if (uiInLineOfSight && !m_creature->IsNonMeleeSpellCasted(false))
```

Comment:
`if (bool uiInLineOfSight = m_creature->IsWithinLOSInMap(m_creature->getVictim())`

If you change this line to that, and wrap line 1379 through 1382 in a new if statement like
`if  (!m_creature->IsNonMeleeSpellCasted(false))`

then you can change the else if into a simple else, and you'll be rid of line 1375 altogether.